homeassistant/components/unifi/__init__.py

```diff
@@ -1,13 +1,32 @@
 """Support for devices connected to UniFi POE."""
+import voluptuous as vol
+
 from homeassistant.const import CONF_HOST
 from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
 
-from .const import CONF_CONTROLLER, CONF_SITE_ID, CONTROLLER_ID, DOMAIN
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    CONF_CONTROLLER, CONF_DETECTION_TIME, CONF_SITE_ID, CONF_SSID_FILTER,
+    CONTROLLER_ID, DOMAIN, UNIFI_CONFIG)
 from .controller import UniFiController
 
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(CONF_DETECTION_TIME): vol.All(
+            cv.time_period, cv.positive_timedelta),
+        vol.Optional(CONF_SSID_FILTER): vol.All(cv.ensure_list, [cv.string])
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
 
 async def async_setup(hass, config):
     """Component doesn't support configuration through configuration.yaml."""
+    hass.data[UNIFI_CONFIG] = []
+
+    if DOMAIN in config:
+        hass.data[UNIFI_CONFIG] = config[DOMAIN]
```

Comment:
Above we set a list and here we set a dict. It should probably be the same type in both cases, right?

Reply:
Right, that's a remnant from when I supported a list of configurations